crypto/rand/drbg_lib.c

```diff
@@ -1079,6 +1084,7 @@ static int drbg_add(const void *buf, int num, double randomness)
     if (num < 0 || randomness < 0.0)
         return 0;
 
+    rand_drbg_lock(drbg);
```

Comment:
Oh, thanks for catiching this one, eagle eye! I missed to move the lock upwards when I added https://github.com/openssl/openssl/commit/8817215d5c52a76f2b184b624bde4df8556dee6d#diff-9181ac017a6177a5f2619f65c9b7a346.